src/django_components/slots.py

```diff
@@ -432,36 +518,57 @@ def render(self, context: Context, name: str, **kwargs: Any) -> SafeString:
 
         # For the user-provided slot fill, we want to use the context of where the slot
         # came from (or current context if configured so)
-        used_ctx = self._resolve_slot_context(context, slot_fill)
+        used_ctx = self._resolve_slot_context(context, slot_fill, component_ctx)
         with used_ctx.update(extra_context):
             # Required for compatibility with Django's {% extends %} tag
             # This makes sure that the render context used outside of a component
             # is the same as the one used inside the slot.
             # See https://github.com/django-components/django-components/pull/859
-            render_ctx_layer = used_ctx.render_context.dicts[-2] if len(used_ctx.render_context.dicts) > 1 else {}
+            if len(used_ctx.render_context.dicts) > 1 and "block_context" in used_ctx.render_context.dicts[-2]:
+                render_ctx_layer = used_ctx.render_context.dicts[-2]
+            else:
+                # Otherwise we simply re-use the last layer, so that following logic uses `with` in either case
+                render_ctx_layer = used_ctx.render_context.dicts[-1]
+
             with used_ctx.render_context.push(render_ctx_layer):
-                # Render slot as a function
-                # NOTE: While `{% fill %}` tag has to opt in for the `default` and `data` variables,
-                #       the render function ALWAYS receives them.
-                output = slot_fill.slot(used_ctx, kwargs, slot_ref)
+                with add_slot_to_error_message(component_name, slot_name):
```

Comment:
By wrapping this in `add_slot_to_error_message`, we get to see the slot path in the error message too.